Scripts/RemediationScripts/Remediate-EnableSSLDBForMySQLFlexibleServer.ps1

```diff
@@ -0,0 +1,801 @@
+            }
+
+            Write-Host "Fetching all Azure Database for MySQL flexible server(s) from [$($FilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $DBForMySQLFSResources = Import-Csv -LiteralPath $FilePath
+
+            $validDBForMySQLFSResources = $DBForMySQLFSResources | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+
+            $validDBForMySQLFSResources | ForEach-Object {
+                $resourceId = $_.ResourceId
+
+                try {                
+                    $DBForMySQLFSResource = Get-AzMySqlFlexibleServer -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
+            
+                    $parameterValue = (Get-AzMySqlFlexibleServerConfiguration -Name $[Constants]::ParameterName -ResourceGroupName $_.ResourceGroupName -ServerName $_.ResourceName -SubscriptionId $SubscriptionId).Value
+                    $DBForMySQLFlexibleServerDetails += $_  | Select-Object @{N = 'ResourceId'; E = { $_.Id } },
+                    @{N = 'ResourceGroupName'; E = { $_.Id.Split("/")[4] } },
+                    @{N = 'ResourceName'; E = { $_.Name } }, 
+                    @{N = 'RequireSecureTransport'; E = { $parameterValue } }
+            
+                }
+                catch {
+                    Write-Host "Error fetching Azure Database for MySQL flexible server(s) resource: Resource ID:  [$($resourceId)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                }
+            }
+        }
+                                                                
+    }
+    
+    $totalDBForMySQLFS = ($DBForMySQLFlexibleServerDetails | Measure-Object).Count
+
+    if ($totalDBForMySQLFS -eq 0) {
+        Write-Host "No Azure Database for MySQL flexible server(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+  
+    Write-Host "Found [$($totalDBForMySQLFS)] Azure Database for MySQL flexible server(s)." -ForegroundColor $([Constants]::MessageType.Update)
+                                                                          
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list for storing Azure Database for MySQL flexible server(s) for which server parameter require_secure_transport in not ON
+     $DBForMySQLFSWithSSLNotEnabled = @()
+
+    Write-Host "Separating Azure Database for MySQL flexible server(s) for which SSL is not enabled..." -ForegroundColor $([Constants]::MessageType.Info)
+
+    $DBForMySQLFlexibleServerDetails | ForEach-Object {
+        if (-not (CheckIfSSLConfigured($_.RequireSecureTransport))) {
+             $DBForMySQLFSWithSSLNotEnabled += $_
+        }
+    }
+   
+     $totalDBForMySQLFSWithSSLNotEnabled = ( $DBForMySQLFSWithSSLNotEnabled  | Measure-Object).Count
+
+    if ( $totalDBForMySQLFSWithSSLNotEnabled -eq 0) {
```

Comment:
Here, we need to output the skipped resources to the log file. Otherwise, in log file there will be 0 skipped resources, which is not the case.

Reply:
updated